### ISSUE TITLE
ngx-echarts : Chart not getting shown on parent display change

### ISSUE THREAD

**Issue by dileepyelleti** [xieziyu/ngx-echarts#209]
If a chart is embedded in a component which is in hidden state (display: none) then chart will not be shown even after changing parent display property. Below is the stackblitz sample to reproduce. Chart will be shown if parent component is shown initially. But it will not be shown if it is hidden and then changed to show.

Stackblitz example:
https://stackblitz.com/edit/angular-echarts-issue

**Comment by xieziyu** [comment]
@dileepyelleti 
It seems echarts get 0 width of its container div, when the parent element is hidden. If you set the width manually, the chart should be shown correctly.

**Comment by dileepyelleti** [comment]
@xieziyu 
I can see this issue happening only with ngx-echarts. With echarts it is working fine. Below code seems to be working.
`<!doctype html>
<html>

<head>
  <title>ECharts Sample</title>
  <script src="http://echarts.baidu.com/dist/echarts.min.js"></script>
</head>

<body>
  <div id="chartParent" style="display: none;">
    <div id="chart" style="width: 500px; height: 350px;"></div>
  </div>
  <button onclick="showChart()">showChart</button>
  <button onclick="hideChart()">hideChart</button>
  <script>
    function showChart() {
      document.getElementById('chartParent').style.display = "block";
    }

    function hideChart() {
      document.getElementById('chartParent').style.display = "none";
    }

    var chart = document.getElementById('chart');
    var myChart = echarts.init(chart);
    var option = {
      title: { text: 'ECharts Sample' },
      tooltip: {},
      legend: { data: ['Sales'] },
      xAxis: { data: ["shirt", "cardign", "chiffon shirt", "pants", "heels", "socks"] },
      yAxis: {},
      series: [{
        name: 'Sales',
        type: 'bar',
        data: [5, 20, 36, 10, 10, 20]
      }]
    };
    myChart.setOption(option);
  </script>
</body>
</html>`

**Comment by xieziyu** [comment]
@dileepyelleti 
You can try *ngIf instead of (display: none)

**Comment by xieziyu** [comment]
@dileepyelleti 
Yes, loading property only works after chart is initialized. In your demo, you set `loadingData` to true before the chart instance is constructed (because of *ngIf).

So you can use (chartInit) to set loadingData. For example:

```html
<div echarts (chartInit)="onChartInit()" [options] ="chartOptions" [initOpts]="initOpts" [merge]="mergeOptions" [loading]="loadingData"></div>
```

```ts
onChartInit() {
  this.loadingData = true;
}
```

**Comment by dileepyelleti** [comment]
@xieziyu 
Even with that approach still facing issue. As we are using apollo graphql for service requests it will cache response and next time it will return immediately. In that case loading is not getting hidden.
Please see my demo having timeout of 0.

**Comment by xieziyu** [comment]
@dileepyelleti 
You can make your service request in `onChartInit()` method instead of `onClick`. Then we can keep `loadingData` value change in order.
